src/BlockTypes/ProductImageGallery.php

```diff
@@ -0,0 +1,63 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+use Automattic\WooCommerce\Blocks\Utils\StyleAttributesUtils;
+
+
+/**
+ * ProductImageGallery class.
+ */
+class ProductImageGallery extends AbstractBlock {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'product-image-gallery';
+
+	/**
+	 *  Register the context
+	 *
+	 * @var string
+	 */
+	protected function get_block_type_uses_context() {
+		return [ 'query', 'queryId', 'postId' ];
+	}
+
+
+	/**
+	 * Include and render the block.
+	 *
+	 * @param array    $attributes Block attributes. Default empty array.
+	 * @param string   $content    Block content. Default empty string.
+	 * @param WP_Block $block      Block instance.
+	 * @return string Rendered block type output.
+	 */
+	protected function render( $attributes, $content, $block ) {
+
+		$post_id = $block->context['postId'];
+		global $product;
```

Comment:
Is there any specific reason behind defining `$product` as global here?

Reply:
Yep. It is necessary, otherwise, `woocommerce_show_product_images();` doesn't work.

Reply:
Got it. Thanks for explaining it to me 🤝